models/repo_branch.go

```diff
@@ -52,6 +63,128 @@ func (repo *Repository) GetBranches() ([]*Branch, error) {
 	return GetBranchesByPath(repo.RepoPath())
 }
 
+// CheckBranchName validates branch name with existing repository branches
+func (repo *Repository) CheckBranchName(name string) error {
+	gitRepo, err := git.OpenRepository(repo.RepoPath())
+	if err != nil {
+		return err
+	}
+
+	if _, err := gitRepo.GetTag(name); err == nil {
+		return ErrTagAlreadyExists{name}
+	}
+
+	branches, err := repo.GetBranches()
+	if err != nil {
+		return err
+	}
+
+	for _, branch := range branches {
```

Comment:
Instead of doing this, how about something like this that scales to thousands of branches?
```
if _, err := gitRepo.GetBranch(name); err != nil {
  return ErrConflict(name)
}
if strings.Contains(name, "/") {
  for index := strings.LastIndex(name, "/"); index > 0; index := strings.LastIndex(name, "/") {
    if conflict, err := gitRepo.GetBranch(name[:index]); err != nil {
      return ErrConflict(name[:index])
    }
  }
}

Reply:
It is not the same (if `name` is `feat` and there already exists branch `feat/test`) this code won't return error

Reply:
One option is to just try the `git branch ...` command, and see if it succeeds or not. I assume that internally git has an efficient way to check for naming conflicts.

Reply:
I think that `gitRepo.CreateBranch` will throw a collision error no?

Reply:
But then `CheckBranchName` will only provide part of validation (as I understand that that was an initial request to create it at first place to reuse it in other places later). Also currently GetBranches is called on every repo page load (in ctx `RepoAssignment`) so it's usage here anyway won't be that big deal